hypothesis-python/tests/cover/test_stateful.py

```diff
@@ -1248,6 +1248,59 @@ def fail_fast(self, param):
     )
 
 
+@pytest.mark.parametrize(
+    "bundle_names,initial,repr_",
+    [
+        ("a", "ret1", "a_0 = state.init()"),
+        ("aba", "ret1", "a_0 = b_0 = a_1 = state.init()"),
+        ("a", multiple(), "state.init()"),
+        ("aba", multiple(), "state.init()"),
+        ("a", multiple("ret1"), "a_0, = state.init()"),
+        ("aba", multiple("ret1"), "a_0, = b_0, = a_1, = state.init()"),
+        ("a", multiple("ret1", "ret2"), "a_0, a_1 = state.init()"),
+        (
+            "aba",
+            multiple("ret1", "ret2"),
+            "\n".join(  # noqa: FLY002  # no, f-string is not more readable
+                [
+                    "a_0, a_2 = state.init()",
+                    "b_0, b_1 = a_0, a_2",
+                    "a_1, a_3 = a_0, a_2",
+                ]
+            ),
+        ),
```

Comment:
I think the ideal here is

```
a_0, a_1 = state.init()
b_0, b_1 = a_0, a_1
a_2, a_3 = a_0, a_1
```

, iterating in order `aabbaa` instead of `abaaba`. Swapping the iteration here to iterate over `rule.targets` first before `result.values` might do the trick:

```python
                        if isinstance(result, MultipleResults):
                            for single_result in result.values:
                                machine._add_result_to_targets(
                                    rule.targets, single_result
                                )
```

(I'm definitely not suggesting adding more work to this PR, I just got rabbit-holed into this)

Reply:
Agree it would look nicer, and I don't mind the (minimal) extra work, _if_ we're confident that changing the storage order is harmless. I guess it should be, since bundles are randomly drawn from?